### PR TITLE
Fix the bug where if id data is updated on the wrong account

### PR DESCRIPTION
**Pull request by qingqingliu**
@bhariharan @wmathurin review please

Fix a bug where existing flow is doing the following thing instead 

1) Call initiateIdentityDataRetrieval with Account A
2) Before ID data returns, switch to Account B
3) ID data returns for account A
4) Mobile SDK updates Account B with ID data for Account A, which is the bug

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.m [hunk 1]
```diff
@@ -134,11 +135,23 @@ - (void)sendRequest
     [request setHTTPShouldHandleCookies:NO];
     [self log:SFLogLevelDebug format:@"SFIdentityCoordinator:Starting identity request at %@", self.credentials.identityUrl.absoluteString];
     
+    __weak __typeof(self) weakSelf = self;
```
Comment (qingqingliu) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.m]: declare weakSelf at top

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.m [hunk 2]
```diff
@@ -134,11 +135,23 @@ - (void)sendRequest
...
+    
     self.session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration ephemeralSessionConfiguration]];
     [[self.session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+         __strong __typeof(self) strongSelf = weakSelf;
```
Comment (qingqingliu) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.m]: declare strong self at the top

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.m [hunk 3]
```diff
@@ -134,11 +135,23 @@ - (void)sendRequest
...
+            [strongSelf log:SFLogLevelDebug format:@"SFIdentityCoordinator Ignore ID data due to account change, old identity URL %@, new identity URL %@",
+             request.URL.absoluteString,
+             currentCredentials.identityUrl.absoluteString];
+            return;
+        }
+        
         if (error) {
-            [self log:SFLogLevelDebug format:@"SFIdentityCoordinator session failed with error: %@", error];
-            [self notifyDelegateOfFailure:error];
+            [strongSelf log:SFLogLevelDebug format:@"SFIdentityCoordinator session failed with error: %@", error];
```
Comment (qingqingliu) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.m]: use strongSelf

**Comment by bhariharan** [comment]
👍

**Pull request merged**